_drafts/CocoaPods-Stats.markdown

```diff
@@ -0,0 +1,99 @@
+
+{% breaking_image /assets/blog_img/stats/stack.png, https://cocoapods.org/pods/ORStackView,  width="1024", no-bottom-margin %}
+
+### Alright, hold up
+
+Let's go over how we check which pods get sent up for analytics, and how we do the unique installs. [CocoaPods-Stats](https://github.com/cocoapods/cocoapods-stats) is a plugin that will be bundled with CocoaPods iwthin a version or two. It [registers](https://github.com/CocoaPods/cocoapods-stats/blob/0361f29ae37e82ccf385319bba9cf31464049144/lib/cocoapods_plugin.rb#L6) as a post-install plugin and is ran on every `pod install` or `pod update`.
+
+#### Detecting Public Pods
+
+ We're very pessimistic about sending a pod up to our [stats server](https://github.com/cocoapods/stats.cocoapods.org) the current code will ensure that you have a CocoaPods/Specs repo set up as your master repo, then ensure that each pod is inside that repo before accepting it as a public domain pod. It looks like this:
+
+ ```ruby
+  # Does the master specs repo exist?
+  master_specs_repo = File.expand_path "~/.cocoapods/repos/master"
+  return unless File.directory? master_specs_repo
+
+  # Is the master specs repo actually the CocoaPods OSS one?
+  Dir.chdir master_specs_repo do
+    git_remote_details = `git remote -v`
+    return unless git_remote_details.include? "CocoaPods/Specs"
+  end
+
+ ...
+
+  # Verify that the pod exists in the master specs repo
+  pods = root_specs.select do |spec|
+    File.directory? File.join(master_specs_repo, "Specs", spec.name)
+  end
+ ```
+
+####  Data being sent
+
+First up, we don't want to know anything about your app. So in order to know unique targets we use your project's target [UUID](https://github.com/artsy/eigen/blob/aea7af93daffb716ccee9aa50ce599dc7949c42b/Artsy.xcodeproj/project.pbxproj#L3888) as an identifier. These are a [hash](http://danwright.info/blog/2010/10/xcode-pbxproject-files-3/) of your MAC address, Xcode's process id and the time of target creation. These UUIDs never change in a project's lifetime. We double hash it just to be super safe.
+
+``` ruby
+  # Grab the project and the type of target
+  uuid = target.user_target_uuids.first
+  project_target = project.objects_by_uuid[uuid]
+
+  # Send in a digested'd UUID anyway, a second layer
+  # of misdirection can't hurt
+  {
+    :uuid => Digest::SHA256.hexdigest(uuid),
+    :type => project_target.product_type,
+    :pods => pods
+  }
+```
+
+We then also send along the CocoaPods version that was used to generate this installation. Ideally before release we'll also be able to give statistics on how many people are using `pod try [pod]` for your library too.
+
+## FAQ
+
+#### Can I opt out as a CocoaPods user?
+
+Sure. You can set an [environment variable](http://apple.stackexchange.com/questions/106778/how-do-i-set-environment-variables-on-os-x) `DISABLE_COCOAPODS_STATS` to true in your shell, and stats will not be sent from your machine.
```

Comment:
For consistency, I think this env variable should be prefixed with COCOAPODS instead of it being in the middle. So `COCOAPODS_DISABLE_STATS`.